src/main/java/app/bpartners/api/service/subscription/StripeSessionFactory.java

```diff
@@ -0,0 +1,152 @@
+import static com.stripe.param.checkout.SessionCreateParams.Mode.SETUP;
+import static com.stripe.param.checkout.SessionCreateParams.Mode.SUBSCRIPTION;
+import static com.stripe.param.checkout.SessionCreateParams.UiMode.HOSTED;
+
+import app.bpartners.api.endpoint.rest.model.RedirectionStatusUrls;
+import app.bpartners.api.model.subscription.Subscription;
+import app.bpartners.api.model.subscription.SubscriptionProduct;
+import com.stripe.exception.StripeException;
+import com.stripe.model.Customer;
+import com.stripe.model.Price;
+import com.stripe.model.SubscriptionSchedule;
+import com.stripe.model.checkout.Session;
+import com.stripe.param.SubscriptionScheduleCreateParams;
+import com.stripe.param.checkout.SessionCreateParams;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import lombok.RequiredArgsConstructor;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+@Slf4j
+public class StripeSessionFactory {
+  public Session createSessionSubscription(
+      Customer stripeCustomer,
+      SubscriptionProduct subscriptionProduct,
+      Price newVariableProductPrice,
+      RedirectionStatusUrls redirectionUrls,
+      Long billingCycleAnchor)
+      throws StripeException {
+    return Session.create(
+        SessionCreateParams.builder()
+            .setMode(SUBSCRIPTION)
+            .setCustomer(stripeCustomer.getId())
+            .setCurrency(defaultCurrency())
+            .addLineItem(
+                SessionCreateParams.LineItem.builder()
+                    .setQuantity(1L)
+                    .setPriceData(
+                        SessionCreateParams.LineItem.PriceData.builder()
+                            .setProduct(subscriptionProduct.getE2Id())
+                            .setCurrency(defaultCurrency())
+                            .setUnitAmount(subscriptionProduct.getPriceInCents())
+                            .setRecurring(
+                                computeRecurringFromSubscriptionProductForSubscriptionMode(
+                                    subscriptionProduct))
+                            .build())
+                    .build())
+            .addLineItem(
+                SessionCreateParams.LineItem.builder()
+                    .setPrice(newVariableProductPrice.getId())
+                    .build())
+            .setSuccessUrl(redirectionUrls.getSuccessUrl())
+            .setCancelUrl(redirectionUrls.getFailureUrl())
+            .setUiMode(HOSTED)
+            .setSubscriptionData(
+                SessionCreateParams.SubscriptionData.builder()
+                    .setProrationBehavior(
+                        SessionCreateParams.SubscriptionData.ProrationBehavior.NONE)
+                    .setBillingCycleAnchor(billingCycleAnchor)
+                    .build())
+            .build());
+  }
+
+  public Session createSessionSetUp(
```

Comment:
Must be private, as it is only called inside the Factory. 

Change the unit test to handle it as well.